cpp/include/cudf/table/experimental/row_operators.cuh

```diff
@@ -1484,12 +1493,12 @@ class device_row_hasher {
   CUDF_HOST_DEVICE device_row_hasher(Nullate check_nulls,
                                      table_device_view t,
                                      uint32_t seed = DEFAULT_HASH_SEED) noexcept
-    : _table{t}, _seed(seed), _check_nulls{check_nulls}
+    : _check_nulls{check_nulls}, _table{t}, _seed(seed)
   {
   }
 
-  table_device_view const _table;
   Nullate const _check_nulls;
+  table_device_view const _table;
```

Comment:
Just curious, is there a style guideline or other reasoning for the ordering change?

Reply:
I'm guessing one of two possibilities:
- He just chose to alphabetize.
- The order in which members are initialized is based on the order that they are declared here, not the order that they appear in the initializer list (the part after the `:` in the constructor). Since `check_nulls` comes before `t` in the constructor signature, he may have reordered the initializer list to match, and then reordering this bit becomes necessary to avoid creating an asymmetry that could catch unwary developers off guard (there are subtle bugs that can come from the wrong initialization order if the constructor makes some invalid assumptions).

Reply:
This is about initialization order matching member order. Compilers sometimes throw warnings about this, and it’s good practice to make the constructor argument order match the initialization order and member order when possible.